test-container/src/main/java/io/strimzi/StrimziKafkaCluster.java

```diff
@@ -0,0 +1,143 @@
+        }
+
+        this.brokersNum = brokersNum;
+        this.network = Network.newNetwork();
+
+        this.zookeeper = new StrimziZookeeperContainer()
+            .withNetwork(this.network)
+            .withNetworkAliases("zookeeper")
+            .withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(StrimziZookeeperContainer.ZOOKEEPER_PORT));
+
+        Map<String, String> defaultKafkaConfigurationForMultiNode = new HashMap<>();
+        defaultKafkaConfigurationForMultiNode.put("offsets.topic.replication.factor", String.valueOf(internalTopicReplicationFactor));
+        defaultKafkaConfigurationForMultiNode.put("num.partitions", String.valueOf(internalTopicReplicationFactor));
+        defaultKafkaConfigurationForMultiNode.put("transaction.state.log.replication.factor", String.valueOf(internalTopicReplicationFactor));
+        defaultKafkaConfigurationForMultiNode.put("transaction.state.log.min.isr", String.valueOf(internalTopicReplicationFactor));
+
+        additionalKafkaConfiguration.putAll(defaultKafkaConfigurationForMultiNode);
+
+        // multi-node set up
+        this.brokers = IntStream
+            .range(0, this.brokersNum)
+            .mapToObj(brokerId -> {
+                LOGGER.info("Starting broker with id {}", brokerId);
+                // adding broker id for each kafka container
+                additionalKafkaConfiguration.put("broker.id", String.valueOf(brokerId));
+
+                return new StrimziKafkaContainer(imageVersion, additionalKafkaConfiguration)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerId)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + StrimziZookeeperContainer.ZOOKEEPER_PORT);
+            })
+            .collect(Collectors.toList());
+    }
+
+    public StrimziKafkaCluster(Map<String, String> additionalKafkaConfiguration) {
+        this(StrimziKafkaContainer.getStrimziVersion() + "-kafka-" + StrimziKafkaContainer.getLatestKafkaVersion(),
+            3,
+            3,
+            additionalKafkaConfiguration);
+    }
+
+    public Collection<StrimziKafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(StrimziKafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    @Override
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(Startable.class::cast);
+        try {
+            Startables.deepStart(startables).get(60, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
+        }
+
+        TestUtils.waitFor("Zookeeper readiness", Duration.ofSeconds(1).toMillis(), Duration.ofSeconds(30).toMillis(),
```

Comment:
We're not waiting for "Zookeeper readiness" here. We're waiting for the broker znode.